library/objective-c/EnvoyConfiguration.m

```diff
@@ -198,11 +202,29 @@ - (nullable NSString *)resolveTemplate:(NSString *)templateYAML {
                             self.appVersion, self.appId];
   [definitions appendFormat:@"- &virtual_clusters %@\n", self.virtualClusters];
 
+  [definitions
+      appendFormat:@"- &stats_flush_interval %lus\n", (unsigned long)self.statsFlushSeconds];
+
+  NSMutableArray *stat_sinks = [NSMutableArray init];
+  [stat_sinks addObjectsFromArray:self.extraStatSinks];
+
   if (self.grpcStatsDomain != nil) {
     [definitions appendFormat:@"- &stats_domain %@\n", self.grpcStatsDomain];
-    [definitions
-        appendFormat:@"- &stats_flush_interval %lus\n", (unsigned long)self.statsFlushSeconds];
     [definitions appendString:@"- &stats_sinks [ *base_metrics_service ]\n"];
+    [stat_sinks addObject:@"*base_metrics_serivce"];
+  }
+
+  bool first = true;
+  if ([stat_sinks count] > 0) {
+    [definitions appendString:@"- &stats_sinks ["];
+
+    for (id sink in stat_sinks) {
+      if (!first) {
```

Comment:
this is never true?

Reply:
Can we check whether `definitions` is empty instead of using a  `first` variable?

Reply:
actually, I know that you asked me for this but I just realized that objective-c has an API for joining all elements of an array with a given separator. Try:
```
[array componentsJoinedByString:@","]
```